### PR TITLE
Adding AzureRM service connection to vstest task

### PR DESCRIPTION
**Pull request by bhaveshbhati**
**Task name**: VsTestV3

**Description**: Added a new ConnectedServiceName input. It allows users to select Azure RM service connection. Once user selects the service connection with Workload Identity Federation as auth scheme it will set the following environment variables 
`AZURESUBSCRIPTION_SERVICE_CONNECTION_ID,AZURESUBSCRIPTION_CLIENT_ID and AZURESUBSCRIPTION_TENANT_ID`

These environment variables can be used in AzurePipelinesCredential provided by Identity SDK.

For example: 

```
string systemAccessToken = Environment.GetEnvironmentVariable("SYSTEM_ACCESSTOKEN");
TokenCredential tokenCredential = new AzurePipelinesCredential(
                    tenantId: Environment.GetEnvironmentVariable("AZURESUBSCRIPTION_TENANT_ID"),
                    clientId: Environment.GetEnvironmentVariable("AZURESUBSCRIPTION_CLIENT_ID"),
                    serviceConnectionId: Environment.GetEnvironmentVariable("AZURESUBSCRIPTION_SERVICE_CONNECTION_ID"),
                    systemAccessToken: systemAccessToken);
```

**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y/N) <Please add link to related issue here>

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

ServiceConnection

### CODE REVIEW

**Comment by bhaveshbhati** [comment]
@microsoft-github-policy-service agree [company="Microsoft"]

**Comment by bhaveshbhati** [comment]
@microsoft-github-policy-service agree

File: Tasks/VsTestV3/runvstest.ts [hunk 1]
```diff
@@ -190,6 +193,25 @@ function isServerBasedRun(): boolean {
     return false;
 }
 
+function setUpConnectedServiceEnvironmentVariables() {
+    var connectedService = tl.getInput('ConnectedServiceName');
+    if(connectedService) {
+        var authScheme: string = tl.getEndpointAuthorizationScheme(connectedService, false);
+        if (authScheme && authScheme.toLowerCase() == "workloadidentityfederation") {
+            process.env.AZURESUBSCRIPTION_SERVICE_CONNECTION_ID = connectedService;
```
Comment (onetocny) [inline Tasks/VsTestV3/runvstest.ts:201]: Maybe you also would like to remove these values from environment once the task completes. Similar to what is done in AzureCLI 2 [here](https://github.com/microsoft/azure-pipelines-tasks/blob/ebb77fe381b29bccd2e0cab033f25ca8df7a62d8/Tasks/AzureCLIV2/azureclitask.ts#L180-L185).

**Review by onetocny — COMMENTED** [review]
In general it looks good to me. I just left one suggestion.

**Review by manolerazvan — APPROVED** [review]

File: Tasks/VsTestV3/runvstest.ts (hunk 1 above)
Comment (bhaveshbhati) [reply Tasks/VsTestV3/runvstest.ts:201]: I was under the impression that the environment variables are not shared across tasks. I did validate that via a sample yaml.

```
- task: VSTest@3
      inputs:
        azureSubscription: 'TestServiceConnection'
        testSelector: 'testAssemblies'
        testAssemblyVer2: |
          **\bin\**\UnitTestProject1.dll
        searchFolder: '$(System.DefaultWorkingDirectory)'
        runSettingsFile: 'cc.runsettings'
        codeCoverageEnabled: true
        platform: '$(buildPlatform)'
        configuration: '$(buildConfiguration)'
      env:
        SYSTEM_ACCESSTOKEN: $(System.AccessToken)

- task: VSTest@3
      inputs:
        testSelector: 'testAssemblies'
        testAssemblyVer2: |
          **\bin\**\UnitTestProject1.dll
        searchFolder: '$(System.DefaultWorkingDirectory)'
        runSettingsFile: 'cc.runsettings'
        codeCoverageEnabled: true
        platform: '$(buildPlatform)'
        configuration: '$(buildConfiguration)'
      env:
        SYSTEM_ACCESSTOKEN: $(System.AccessToken)
```
Second task failed saying that AZURESUBSCRIPTION_SERVICE_CONNECTION_ID is NULL.

**Review by onetocny — APPROVED** [review]

**Pull request merged**